machine_learning/labeling/image_providers.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import cv2
+
+class BaseImageProvider(object):
+    '''Classes that provide image sources. Examples may include:
+        - A bag crawler
+        - A video crawler
+        - Loading images from a folder
+    '''
+    name = "general image provider"
+
+    def load_source(self, *args, **kwargs):
+        ''' Load images from source '''
+        pass
+
+    def get_next_image(self):
```

Comment:
In reference to the above comment, only the labeler uses this function right now, so no need until it's exposed.